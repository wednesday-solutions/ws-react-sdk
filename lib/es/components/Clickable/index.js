import{_ as e}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"../_rollup-plugin-inject-process-env-0dfb24f4.js";import r from"react";import t from"prop-types";import o from"styled-components";import i from"../T/index.js";import"react-intl";import"../If/index.js";import"../fonts-56bad7cc.js";import"../media-54d0e07e.js";var n,p=o.div(n||(n=e(["\n  color: #1890ff;\n  &:hover {\n    cursor: pointer;\n  }\n"])));function s(e){var{onClick:t,textId:o}=e;return r.createElement(p,{"data-testid":"clickable",onClick:t},o&&r.createElement(i,{id:o}))}s.propTypes={onClick:t.func.isRequired,textId:t.string.isRequired};export default s;
