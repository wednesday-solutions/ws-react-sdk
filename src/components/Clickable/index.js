/**
 *
 * Clickable
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import T from '@components/T';

const StyledClickable = styled.div`
  color: #1890ff;
  &:hover {
    cursor: pointer;
  }
`;
function Clickable({ onClick, textId }) {
  return (
    <StyledClickable data-testid="clickable" onClick={onClick}>
      {textId && <T id={textId} />}
    </StyledClickable>
  );
}

Clickable.propTypes = {
  onClick: PropTypes.func.isRequired,
  textId: PropTypes.string.isRequired
};

export default Clickable;
