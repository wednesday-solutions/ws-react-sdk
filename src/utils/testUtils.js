import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
