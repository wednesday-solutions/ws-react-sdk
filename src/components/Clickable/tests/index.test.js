/**
 *
 * Tests for Clickable
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import Clickable from '../index';

describe('<Clickable /> component tests', () => {
  let clickSpy;
  beforeEach(() => {
    clickSpy = jest.fn();
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Clickable onClick={clickSpy} textId="repo_list" />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 Clickable component', () => {
    const { getAllByTestId } = renderWithIntl(<Clickable onClick={clickSpy} textId="repo_list" />);
    expect(getAllByTestId('clickable').length).toBe(1);
  });
});
