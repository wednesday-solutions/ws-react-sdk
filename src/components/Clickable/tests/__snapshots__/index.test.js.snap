// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clickable /> component tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gtsrHT eyZGKz"
      data-testid="clickable"
    >
      <p
        class="sc-bdnxRM caWfaj"
        data-testid="t"
      >
        repo_list
      </p>
    </div>
  </div>
</body>
`;
